citesphere-importer/src/main/java/edu/asu/diging/citesphere/importer/core/service/parse/impl/CrossRefHandler.java

```diff
@@ -0,0 +1,39 @@
+package edu.asu.diging.citesphere.importer.core.service.parse.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.citesphere.importer.core.exception.HandlerTestException;
+import edu.asu.diging.citesphere.importer.core.exception.IteratorCreationException;
+import edu.asu.diging.citesphere.importer.core.service.impl.JobInfo;
+import edu.asu.diging.citesphere.importer.core.service.parse.BibEntryIterator;
+import edu.asu.diging.citesphere.importer.core.service.parse.FileHandler;
+import edu.asu.diging.citesphere.importer.core.service.parse.IHandlerRegistry;
+import edu.asu.diging.citesphere.importer.core.service.parse.iterators.CrossRefIterator;
+import edu.asu.diging.citesphere.importer.core.service.parse.jstor.xml.IArticleTagParser;
+
+@Service
+public class CrossRefHandler implements FileHandler {
+
+    @Autowired
+    private IArticleTagParser parserRegistry;
+    
+    @Value("${_citesphere_download_path}")
+    private String downloadPath;
+
+    @Override
+    public boolean canHandle(String path) throws HandlerTestException {
```

Comment:
this doesn't make sense. What if we want to add another service? The way it is right now, it could only handle one service which will be used when path is null. Instead, the subclasses of the AbstractImportProcessor should only have a method to get a "BibEntryIterator". the one for files will use the filehandler registry, the one for cross reff will just create a crossrefiterator.

Reply:
Removed the class, as it was just being used to get the iterator.